xarray/core/missing.py

```diff
@@ -560,8 +560,8 @@ def _localize(var, indexes_coords):
     indexes = {}
     for dim, [x, new_x] in indexes_coords.items():
         index = x.to_index()
-        imin = index.get_loc(np.min(new_x.values), method="nearest")
-        imax = index.get_loc(np.max(new_x.values), method="nearest")
+        imin = index.get_loc(np.nanmin(new_x.values), method="nearest")
+        imax = index.get_loc(np.nanmax(new_x.values), method="nearest")
```

Comment:
It looks that np.nanmin (and nanmax) supports np.datetime-dtype only with numpy>=1.18.
We can copy [np.nanmin](https://github.com/numpy/numpy/blob/v1.19.0/numpy/lib/nanfunctions.py#L228-L336) to our core/npcompat.py and call this function here.

Reply:
I think they added some low level loops to fix this (numpy/numpy#14841) so guess we cannot copy nanmin over and have to use `if LooseVersion(np. __version__) < LooseVersion("1.18")` (or similar).